sky/packages/sky/lib/src/material/bottom_sheet.dart

```diff
@@ -101,9 +133,17 @@ class _ModalBottomSheetRoute extends PerformanceRoute {
     );
   }
 
+  void didPush(NavigatorState navigator) {
+    super.didPush(navigator);
+    _performance?.forward();
+  }
+
   void didPop([dynamic result]) {
     completer.complete(result);
-    super.didPop(result);
+    if (_performance.status != PerformanceStatus.dismissed) {
+      _performance?.reverse();
+      super.didPop(result);
```

Comment:
As I understand it _handlePerformanceChanged in Route handles the Performance being dismissed by removing the route. Calling super.didPop() in that case fails.